projects/graphql/example/tests/custom-rule-test.js

```diff
@@ -0,0 +1,78 @@
+const { expect } = require('chai');
+const { RuleTester } = require('eslint');
+const ruleTester = new RuleTester({
+  parser: '@eslint-ast/eslint-plugin-graphql/parser',
+  parser: require.resolve(`${__dirname}/../../eslint-plugin-graphql/parser`),
+  parserOptions: {
+    schema: `${__dirname}/fixtures/schema.graphql`,
+  },
+});
+
+// This is a very odd way of testing parserServices.pathOf
+// However we do it this way because we don't have parent edges between nodes
+// after parsing: this is added at some other time.  I believe it happens
+// during eslint's run of the visitor pattern for its rules so fully testing
+// parser services w/fully hydrated nodes seems to be best done indirectly via
+// rule testing.
+describe('custom rules', function () {
+  const rule = {
+    type: 'problem',
+    meta: {
+      messages: {
+        msg: 'msg: {{ path }} // {{ pathCustomDelim }}',
+      },
+    },
+    create(context) {
+      return {
+        SelectionSet(node) {
+          if (node.selections.find(f => f.name.value === 'title')) {
+            context.report({
+              node,
+              messageId: 'msg',
+              data: {
+                path: context.parserServices.pathOf(node),
+                pathCustomDelim: context.parserServices.pathOf(node, ' :: '),
+              },
+            });
+          }
+        },
+      };
+    },
+  };
+
+  ruleTester.run('no-book-title', rule, {
+    valid: [
+      {
+        code: `
+        query {
+          bookstores {
+            name
+          }
+        }
+      `,
+      },
+    ],
+    invalid: [
+      {
+        code: `
+        query {
+          bookstores {
+            books {
+              title
+            }
+          }
+        }
+      `,
```

Comment:
Are there any other permutations you think we should be testing for here?

Reply:
hmm there's a couple things we might as well test

1. paths that use variables (shouldn't affect anything but might as well have a test for it)
2. paths that cross fragment boundaries (inline + in-doc)

Reply:
@scalvert I've updated the implementation and added tests to cover more cases; thoughts?

Reply:
👍 tests look good. Thanks for adding.